javascript/ql/src/semmle/javascript/dataflow/TypeTracking.qll

```diff
@@ -265,6 +260,33 @@ class TypeBackTracker extends TTypeBackTracker {
    * This predicate is only defined if the type has not been tracked into a property.
    */
   TypeBackTracker continue() { prop = "" and result = this }
+
+  /**
+   * Gets the summary that corresponds to having taken a backwards step from `succ` to `pred`.
+   */
+  pragma[inline]
+  TypeBackTracker step(DataFlow::SourceNode pred, DataFlow::SourceNode succ) {
+    exists(StepSummary summary |
+      StepSummary::step(pred, succ, summary) and
+      this = result.prepend(summary)
+    )
+  }
+
+  /**
+   * Gets the summary that corresponds to having taken a backwards step from `succ` to `pred`.
+   *
+   * `TypeBackTracker::step` is a less general, but more performant, variant of this predicate.
```

Comment:
This explanation isn't detailed enough.

Reply:
I have pushed a change. Does it contain the kind of detail you want?

Reply:
Perfect, thanks!